src/Orleans.Core/Serialization/TypeUtilities.cs

```diff
@@ -231,12 +242,12 @@ public static bool IsAccessibleFromAssembly(Type type, Assembly assembly)
             }
             else if (typeInfo.IsGenericTypeDefinition)
             {
-                // Guard against invalid type constraints, which appear when generating code for some languages, such as F#.
+                // Guard against unrepresentable type constraints, which appear when generating code for some languages, such as F#.
                 foreach (var parameter in typeInfo.GenericTypeParameters)
                 {
                     foreach (var constraint in parameter.GetTypeInfo().GetGenericParameterConstraints())
                     {
-                        if (IsSpecialClass(constraint)) return false;
+                        if (constraint == typeof(Array) || constraint == typeof(Delegate) || constraint == typeof(Enum)) return false;
```

Comment:
You dropped `ValueType` when inlining. Was it by design?

Reply:
It is: it was supposed to tell us what types cannot be used as generic constraints in C#, but `ValueType` can (`T : struct`).